resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -213,7 +218,6 @@ public DnsNameResolver(
             int ndots,
             boolean decodeIdn) {
         super(eventLoop);
-        checkNotNull(channelFactory, "channelFactory");
```

Comment:
this class doesn't use it or know any of the constraints about this variable. It is used by `Bootstrap` and the logic for what is valid lives in that class (already checks for NPE).